packages/pluggableWidgets/notifications-native/src/Notifications.tsx

```diff
@@ -1,67 +1,139 @@
-        getHandler: (action: ActionsType) => ActionValue | undefined
-    ): void => {
-        const body: string = notification.body ?? "";
-        const title: string = notification.title ?? "";
-        const subtitle: string = notification.ios?.subtitle ?? "";
-        const actions = props.actions.filter(item => item.name === data?.actionName);
-
-        if (actions.length === 0) {
-            return;
-        }
+    const handleNotification = useCallback(
+        (notification: Notification, getHandler: (action: ActionsType) => Option<ActionValue>): void => {
+            const body: string = notification.body ?? "";
+            const title: string = notification.title ?? "";
+            const subtitle: string = notification.subTitle ?? "";
+            const actions = props.actions.filter(item => item.name === notification.data?.actionName);
 
-        if (props.guid) {
-            props.guid.setValue(data?.guid);
-        }
-        if (props.title) {
-            props.title.setValue(title);
-        }
-        if (props.subtitle) {
-            props.subtitle.setValue(subtitle);
-        }
-        if (props.body) {
-            props.body.setValue(body);
-        }
-        if (props.action) {
-            props.action.setValue(actions.join(" "));
-        }
+            if (actions.length === 0) {
+                return;
+            }
+            if (props.guid) {
+                props.guid.setValue(notification.data?.guid);
+            }
+            if (props.title) {
+                props.title.setValue(title);
+            }
+            if (props.subtitle) {
+                props.subtitle.setValue(subtitle);
+            }
+            if (props.body) {
+                props.body.setValue(body);
+            }
+            if (props.action) {
+                props.action.setValue(actions.map(action => action.name).join(" "));
+            }
 
-        actions.forEach(action => {
-            const handler = getHandler(action);
-            executeAction(handler);
-        });
-    };
+            actions.forEach(action => executeAction(getHandler(action)));
+        },
+        [props.actions, props.guid, props.title, props.subtitle, props.body, props.action]
+    );
 
-    const onReceive = (notification: FirebaseMessagingTypes.RemoteMessage): void => {
-        if (notification.notification) {
-            handleNotification(notification.notification, notification.data, action => action.onReceive);
-        }
-    };
+    const remoteMessageHandlerHelpers = (
+        notification: FirebaseMessagingTypes.Notification
+    ): Pick<Notification, "title" | "subTitle" | "body"> => ({
+        title: notification?.title,
+        body: notification?.body,
+        subTitle: notification.ios?.subtitle
+    });
+
+    const onReceive = useCallback(
+        (message: FirebaseMessagingTypes.RemoteMessage): void => {
+            const { notification } = message;
+            if (notification) {
```

Comment:
I would rather use message.notification or de-structure all the elements as you use data in the line 89

Reply:
I don't feel it's necessary

Reply:
I agree with Diego on this one; destructuring both properties allows for shorthand notation on line 89 as well.